lib/datadog.ts

```diff
@@ -63,3 +73,34 @@ export class Datadog extends cdk.Construct {
     }
   }
 }
+
+function validateProps(props: DatadogProps) {
+  const siteList: string[] = ["datadoghq.com", "datadoghq.eu", "us3.datadoghq.com", "ddog-gov.com"];
+  if (props.apiKey !== undefined && props.apiKMSKey !== undefined) {
+    throw new Error("Both `apiKey` and `apiKMSKey` cannot be set.");
+  }
+
+  if (props.site !== undefined && !siteList.includes(props.site.toLowerCase())) {
+    throw new Error(
+      "Warning: Invalid site URL. Must be either datadoghq.com, datadoghq.eu, us3.datadoghq.com, or ddog-gov.com.",
+    );
+  }
+
+  if (props.apiKey === undefined && props.apiKMSKey === undefined && props.flushMetricsToLogs === false) {
+    throw new Error("When `flushMetricsToLogs` is false, `apiKey` or `apiKMSKey` must also be set.");
+  }
+
+  // TODO: Refactor this error when Extension layer officially supports tracing.
+  if (props.enableDDTracing === true && props.forwarderARN === undefined) {
```

Comment:
After thinking about this more, I think we should not add this check. Otherwise it will be impossible to use the extension for tracing until we update this. So, let's leave this out. Sorry for the confusion.